frontend/src/components/organisms/MatchesList/MatchesList.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {useEffect} from 'react';
+import styled from "styled-components";
+import {Card} from '@material-ui/core';
+import {MatchItem} from "../../molecules/MatchItem/MatchItem";
+import {MIN_CARD_COMPONENT_WIDTH} from "../../atoms/constants/sizes";
+import {MatchListItem} from "./MatchListItem";
+import {MatchesListRestApi} from "../../../restapi/matches-list";
+import {MatchesListDto} from "./MatchesListDto";
+import {MatchStatus} from "./MatchStatus";
+
+const StyledMatchesList = styled(Card)({
+    width: MIN_CARD_COMPONENT_WIDTH,
+});
+
+export type MatchesListProps = {
+    readonly tournamentId: string;
+};
+
+export const MatchesList = ({tournamentId}: MatchesListProps) => {
+
+    const [expanded, setExpanded] = React.useState<string | boolean>(false);
+    const [matchesListItems, setMatchesListItems] = React.useState<MatchListItem[] | undefined>();
+
+    useEffect(() => {
+        MatchesListRestApi()
+            .getMatchesList(tournamentId)
+            .then((matchesListDto) => {
+                const newMatchesListItems = createMatchListItemsFromMatchesListDto(matchesListDto);
+                setMatchesListItems(newMatchesListItems)
+            });
+    }, [tournamentId]);
+
+    const handleChangeExpander = (panel: string | boolean) => (event: any, isExpanded: string | boolean) => {
+        setExpanded(isExpanded ? panel : false);
+    };
+
+    return (
+        <>
+            <StyledMatchesList>
+                {matchesListItems ? matchesListItems.map((item, index) => generateMatchItem(item, index, expanded, handleChangeExpander)) : "Oczekiwanie na pobranie"}
+            </StyledMatchesList>
+        </>
+    )
+};
+
+const generateMatchItem = (matchItem: MatchListItem, index: number, expanded: string | boolean, handleChangeExpander: (panel: string | boolean) => (event: any, isExpanded: string | boolean) => void) => (
```

Comment:
Ta funkcja raczej nie ma sensu :) Bo nie robi nic więcej niż przekazywanie dokladnie parametrow i zwracanie komponentu. Łatwo możesz zrobić to tak:
```
<MatchItem
        {...matchItem}
        key={matchItem.matchNumber}
        expanded={expanded}
        handleChangeExpander={handleChangeExpander}
    />
```

Reply:
No właściwie, ale chciałem wyrzucić poza funkcję główną, ale chyba masz rację, że można jednak zrezygnować z takiego wyodrębniania.

Moge jakieś Za i Przeciw?

Reply:
Jak sadzisz :) To generate na poczatku zwrocilo moja uwage, ze ja przeciez tutaj jakos nic nie generuje - po prostu zwracam ten komponent.

Reply:
ok